src/pkg/bundle/deploy.go

```diff
@@ -293,6 +293,26 @@ func (b *Bundle) PreDeployValidationTF() (string, string, string, error) {
 	}
 	b.bundle.Packages = tfHelperConfig.Packages
 
+	// Write a custom .terraformrc that points to the provider we brought
+	customTFRC := tfparser.TerraformRC{}
+	customTFRC.ProviderInstallation.Direct = make(map[string]string)
+	customTFRC.ProviderInstallation.DevOverrides = make(map[string]string)
+	customTFRC.ProviderInstallation.DevOverrides["defenseunicorns/uds"] = b.tmp
+	err = customTFRC.WriteHCL(filepath.Join(b.tmp, config.TerraformRC))
+	if err != nil {
+		return "", "", "", err
+	}
+
+	// Copy the contents of the terraform provider binary into a file in the tmpdir
+	terraformProvider, err := os.ReadFile(filepaths[config.TerraformProvider])
+	if err != nil {
+		return "", "", "", err
+	}
+	err = os.WriteFile(filepath.Join(b.tmp, config.TerraformProvider), terraformProvider, 0777) //nolint:gosec
+	if err != nil {
+		return "", "", "", err
+	}
```

Comment:
nit: if the binary is large-ish then `io.Copy()` might be better here than reading the entire binary first, but that might not be a problem